scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -1077,25 +1083,30 @@ def handle(self):
         )
 
         try:
-            # Worker is responsible for processing the data read from one request.
-            # Using the queue ensures the data is processed in the order it was read.
-            DONE = "DONE"
-            work_queue = queue.Queue()
-            def worker(queue, is_shutdown):
-                data = queue.get(block=True)
-                while data != DONE:
-                    if is_shutdown():
-                        global_log.info("ThreadPool shutting down, skipping further request processing.")
-                        break
-                    self.__request_data_process(syslog_parser, data)
+            if self.__request_processing_executor:
+                # Worker is responsible for processing the data read from one request.
+                # Using the queue ensures the data is processed in the order it was read.
+                DONE = "DONE"
+                work_queue = queue.Queue()
+                def worker(queue, is_shutdown):
                     data = queue.get(block=True)
+                    while data != DONE:
+                        if is_shutdown():
+                            global_log.info("ThreadPool shutting down, skipping further request processing.")
+                            break
+                        self.__request_data_process(syslog_parser, data)
+                        data = queue.get(block=True)
```

Comment:
I would consider rearranging this to avoid the special value & handle queue cleanup during shutdown:

```
def worker(queue, is_shutdown):
    while True:
        if is_shutdown():
            global_log.info("ThreadPool shutting down, skipping further request processing.")
            break
        try:
            data = queue.get(block=False, timeout=1.0)
        except Empty:
            continue
        self.__request_data_process(syslog_parser, data)
        queue.task_done()

    while not queue.empty():
        data = queue.get()
        queue.task_done()
```

Then the producer calls `queue.join()` to ensure everything from the queue has been consumed